sources/packages/backend/apps/api/src/route-controllers/overaward/_tests_/e2e/overaward.students.controller.getOverawardsByStudent.e2e-spec.ts

```diff
@@ -0,0 +1,98 @@
+import { HttpStatus, INestApplication } from "@nestjs/common";
+import * as request from "supertest";
+import {
+  createFakeDisbursementOveraward,
+  createFakeStudentAssessment,
+  createFakeApplication,
+} from "@sims/test-utils";
+import { Repository } from "typeorm";
+import {
+  Application,
+  AssessmentTriggerType,
+  DisbursementOveraward,
+  DisbursementOverawardOriginType,
+  Student,
+  StudentAssessment,
+} from "@sims/sims-db";
+
+import {
+  BEARER_AUTH_TYPE,
+  createTestingAppModule,
+  getStudentToken,
+  FakeStudentUsersTypes,
+  getStudentByFakeStudentUserType,
+} from "../../../../testHelpers";
+import { StudentsOverawardAPIOutDTO } from "../../models/overaward.dto";
+
+describe("OverawardStudentsController(e2e)-getOverawardsByStudent", () => {
+  let app: INestApplication;
+  let assessmentRepo: Repository<StudentAssessment>;
+  let applicationRepo: Repository<Application>;
+  let disbursementOverawardRepo: Repository<DisbursementOveraward>;
+  let student: Student;
+
+  beforeAll(async () => {
+    const { nestApplication, dataSource } = await createTestingAppModule();
+    app = nestApplication;
+    assessmentRepo = dataSource.getRepository(StudentAssessment);
+    applicationRepo = dataSource.getRepository(Application);
+    disbursementOverawardRepo = dataSource.getRepository(DisbursementOveraward);
+    student = await getStudentByFakeStudentUserType(
+      FakeStudentUsersTypes.FakeStudentUserType1,
+      dataSource,
+    );
+  });
+
+  it("Should return student overawards", async () => {
+    // Prepare the student assessment to create overaward.
+    const application = await applicationRepo.save(
+      createFakeApplication({
+        student,
+      }),
+    );
+    const studentAssessment = await assessmentRepo.save(
```

Comment:
just a suggestion, line 53 could have been added before line 48, and `studentAssessment` could have passed as `currentStudentAssessment` in line 49

Reply:
To create the fake assessment before, how do we pass the application object?